Source/WebCore/editing/MarkupAccumulator.cpp

```diff
@@ -262,6 +262,9 @@ void MarkupAccumulator::serializeNodesWithNamespaces(Node& targetNode, Serialize
             }
 
             shouldAppendNode = !(current == &targetNode && root != SerializedNodes::SubtreeIncludingNode);
+            ASSERT(current);
+            if (!current)
+                break;
             shouldEmitCloseTag = !(targetNode.document().isHTMLDocument() && elementCannotHaveEndTag(*current));
```

Comment:
I think the early return should come before calculating both the flags's values:
```
ASSERT(current);
if (!current)
    break;

shouldAppendNode = ...
shouldEmitCloseTag = ...
```